detectable/src/main/java/com/synopsys/integration/detectable/detectables/setuptools/SetupToolsExtractUtils.java

```diff
@@ -0,0 +1,93 @@
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsCfgParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsPyParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsTomlParser;
+
+public class SetupToolsExtractUtils {
+    
+    private static final String BUILD_KEY = "build-system.requires";
+    private static final String REQUIRED_KEY = "setuptools";
+    private static final String TOML_DEPENDENCIES = "project.dependencies";
+    
+    private static final String SETUP_CFG = "setup.cfg";
+    private static final String SETUP_PY = "setup.py";
+
+    public static TomlParseResult extractToml(File projectToml) throws IOException {
+        String projectTomlText = FileUtils.readFileToString(projectToml, StandardCharsets.UTF_8);
+
+        return Toml.parse(projectTomlText);
+    }
+
+    public static boolean checkTomlRequiresSetupTools(TomlParseResult parsedToml) {
+        if (parsedToml != null) {
+            TomlArray buildRequires = parsedToml.getArray(BUILD_KEY);
+
+            if (buildRequires != null) {
+                for (int i = 0; i < buildRequires.size(); i++) {
+                    String requires = buildRequires.getString(i);
+
+                    if (requires.contains(REQUIRED_KEY)) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public static SetupToolsParser findDependenciesFile(TomlParseResult parsedToml, FileFinder fileFinder, DetectableEnvironment environment) throws IOException {
+        // Dependencies, if they exist at all, will be in one of three files.
+        // Step 1: Check the pyproject.toml
+        TomlArray tomlDependencies = parsedToml.getArray(TOML_DEPENDENCIES);
+        
+        if (tomlDependencies != null && tomlDependencies.size() > 0) {
+            return new SetupToolsTomlParser(parsedToml);
+        }
+        
+        // Step 2: Check the setup.cfg
+        Requirements fileResolver = new Requirements(fileFinder, environment);
+        File cfgFile = fileResolver.file(SETUP_CFG);
+        
+        if (cfgFile != null) {
+            SetupToolsCfgParser cfgParser = new SetupToolsCfgParser(parsedToml);
+
+            List<String> cfgDependencies = cfgParser.load(cfgFile.toString());
+
+            if (cfgDependencies != null && cfgDependencies.size() > 0) {
+                return cfgParser;
+            }
+        }
+        
+        // Step 3: Check the setup.py
+        fileResolver = new Requirements(fileFinder, environment);
+        File pyFile = fileResolver.file(SETUP_PY);
+        
+        if (pyFile != null) {
+            SetupToolsPyParser pyParser = new SetupToolsPyParser(parsedToml);
+
+            List<String> pyDependencies = pyParser.load(pyFile.toString());
```

Comment:
Can the pyFile be large enough to impact performance?

Reply:
I want to say the answer is no but it's difficult to definitively say. The setup.py is really flexible as it is code based and not simply a text based config file. Also, there is no reason to assume best practices are followed, etc. Still, it is a text file so I don't think it would be insanely large. I'd caution us changing this area at this point, especially since the setup.py is a bit of a deprecated file and less and less projects will hopefully be using it as time goes on.